app/services/ValidatingPageBuilder.scala

```diff
@@ -165,4 +182,55 @@ class ValidatingPageBuilder @Inject() (pageBuilder: PageBuilder){
         }
     }
 
+  private def checkForSequencePageReuse(connections: List[PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[GuidanceError] = {
+    val vertexMap = connections.map(pv => (pv.id, pv)).toMap
+    val mainFlow: List[String] = mainFlowPageIds(vertexMap)
+    val sequencePageIds = for{
+      pv <- connections.filterNot(_.flows.isEmpty)              // Sequences
+      flw <- pv.flows                                           // Flow Ids
+      p <- pageGraph(findPages(List(flw)), vertexMap, mainFlow) // Pages below sequecnes
+    } yield p.id
+
+    sequencePageIds.groupBy(x => x).toList.collect{case m if m._2.length > 1 => PageOccursInMultiplSequenceFlows(m._1)}
+  }
+
+  private def mainFlowPageIds(vertexMap: Map[String, PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[String] =
+    pageGraph(List(Process.StartStanzaId), vertexMap).map(_.id)
+
+  @tailrec
+  // Given a list of stanza ids, find all connected pages (wont follow links)
+  private def findPages(ids: List[String], seen: List[String] = Nil, acc: List[String] = Nil)(implicit stanzaMap: Map[String, Stanza]): List[String] =
+    ids match {
+      case Nil => acc
+      case x :: xs if seen.contains(x) => findPages(xs, seen, acc)
+      case x :: xs =>
+        stanzaMap(x) match {
+          case _: PageStanza => findPages(xs, x :: seen, x :: acc)
+          case EndStanza => findPages(xs, x :: seen, acc)
+          case s => findPages(s.next.toList.filterNot(seen.contains(_)) ++ xs, x :: seen, acc)
+        }
+    }
+
+  @tailrec
+  // Given a list of page ids, find all connected pages (wont follow links)
+  private def pageGraph(keys: List[String],
+                        vertices: Map[String, PageVertex],
+                        ignore: List[String] = Nil,
+                        dontFollowFlows: Boolean = true,
+                        seen: List[String] = Nil,
+                        acc: List[PageVertex] = Nil)(implicit stanzaMap: Map[String, Stanza]): List[PageVertex] =
+    keys match {
+      case Nil => acc
+      case x :: xs if seen.contains(x) => pageGraph(xs, vertices, ignore, dontFollowFlows, seen, acc)
+      case x :: xs if !ignore.contains(x) =>
+        val v = vertices(x)
+        if (vertices(x).flows.isEmpty || dontFollowFlows) {
```

Comment:
I'd use v in the if condition instead of vertices(x), seeing as you have bothered to define it.

Reply:
done